AI/BattleAI/AttackPossibility.cpp

```diff
@@ -28,6 +28,22 @@ int64_t AttackPossibility::attackValue() const
 	return damageDiff();
 }
 
+int64_t AttackPossibility::calculateDpsReduce(
+	const battle::Unit * attacker,
+	const battle::Unit * defender,
+	uint64_t damageDealt,
+	std::shared_ptr<CBattleInfoCallback> cb)
+{
+	vstd::amin(damageDealt, defender->getAvailableHealth());
+
+	auto enemyDamageBeforeAttack = cb->battleEstimateDamage(BattleAttackInfo(defender, attacker, defender->canShoot()));
+	auto enemiesKilled = damageDealt / defender->MaxHealth() + (damageDealt % defender->MaxHealth() >= defender->getFirstHPleft() ? 1 : 0);
+	auto enemyDps = (enemyDamageBeforeAttack.first + enemyDamageBeforeAttack.second) / 2;
+	auto dpsPerEnemy = enemyDps / (double)defender->getCount();
+
+	return (int64_t)(dpsPerEnemy * (enemiesKilled + damageDealt / (double)defender->MaxHealth()) / 2);
```

Comment:
Average. First and second is min/max

Reply:
damageDealt = 100
defeder->maxHelth() = 25

enemiesKilled = 4 or 5

returned value will be either dpsPerEnemy * 4 or dpsPerEnemy * 4.5

Is it desired result?

Reply:
I suspect that you want to count killing single creature as 1 not as 0.5 (or maybe even more)
Thus, you can return dpsPerEnemy * enemiesKilled

Reply:
Actually it is not that important because afterwards we just search minmax but the idea is that half of killing bounty goes for health and half for actual kill so that hitting enemy without kill does not look totally pointless.